ANMN/LTSP/TSaggregator/aggregated_timeseries.py

```diff
@@ -49,56 +49,57 @@ def good_file(nc, VoI, site_code):
     if LATITUDE is a dimension has length 1
     if LONGITUDE is a dimension has length 1
 
+    :param file: name of the netcdf file
     :param nc: xarray dataset
     :param VoI: string. Variable of Interest
     :param site_code: code of the mooring site
-    :return: boolean
+    :return: dictionary with the file name and list of failed tests
     """
 
     attributes = list(nc.attrs)
     variables = list(nc.variables)
     dimensions = list(nc.dims)
-    VoIdimensions = list(nc[VoI].dims)
     allowed_dimensions = ['TIME', 'LATITUDE', 'LONGITUDE']
+    error_list = []
+    error_dict = {}
 
-    criteria_site = nc.site_code == site_code
-    criteria_FV = 'Level 1' in nc.file_version
-    criteria_TIME = 'TIME' in variables
-    criteria_LATITUDE = 'LATITUDE' in variables
-    criteria_LONGITUDE = 'LONGITUDE' in variables
-    criteria_NOMINALDEPTH = 'NOMINAL_DEPTH' in variables or 'instrument_nominal_depth' in attributes
-    criteria_VoI = VoI in variables
-    criteria_VoIdimensionTIME =  'TIME' in VoIdimensions
-
-    criteria_LAT_VoIdimension = True
-    if 'LATITUDE' in VoIdimensions:
-        if len(nc.LATITUDE) > 1:
-            criteria_LAT_VoIdimension = False
-
-    criteria_LON_VoIdimension = True
-    if 'LONGITUDE' in VoIdimensions:
-        if len(nc.LATITUDE) > 1:
-            criteria_LON_VoIdimension = False
-
-    criteria_alloweddimensions = True
-    for d in range(len(VoIdimensions)):
-        if VoIdimensions[d] not in allowed_dimensions:
-            criteria_alloweddimensions = False
-            break
-
-    all_criteria_passed = criteria_site and \
-                          criteria_FV and \
-                          criteria_TIME and \
-                          criteria_LATITUDE and \
-                          criteria_LONGITUDE and \
-                          criteria_VoI and \
-                          criteria_NOMINALDEPTH and \
-                          criteria_LON_VoIdimension and \
-                          criteria_LAT_VoIdimension and \
-                          criteria_VoIdimensionTIME and \
-                          criteria_alloweddimensions
-
-    return all_criteria_passed
+    if nc.site_code != site_code:
```

Comment:
I would use `getattr(nc, 'site_code', '')`, just in case the attribute is missing. Same for file_version below.

Reply:
good. done.